docs: add guide to use with Docker (#55)
slidevjs/slidev: docs/guide/install.md

> ```bash
> echo 'shamefully-flatten=true' >> .npmrc
> ```

## Install on Docker

> If you need a rapid way to run a presentation, you can use the prebuilt (or even build your own) [docker](https://hub.docker.com/r/stig124/slidev) image (maintained by [stig124](https://github.com/Stig124))

Reviewer: Thanks for managing the migrations. Since we already have the repo with the readme, how about let's keep this section with a link to the repo so we don't need to maintain duplicated docs?

Author: And there is done

> ```bash
> echo 'shamefully-flatten=true' >> .npmrc
> ```

## Install on Docker

If you need a rapid way to run a presentation, you can use the prebuilt (or even build your own) [docker](https://hub.docker.com/r/stig124/slidev) image (maintained by [stig124](https://github.com/Stig124))

> ##### *(this is not meant as a developper environnement, as it uses only npm-fetched binaries)*

Reviewer: ```suggestion
##### *this is not meant to be a developing environment, it uses only npm-fetched distributions*
```

Pull the image :

```bash
$ docker pull stig124/slidev:latest
```

> [podman](https://podman.io/) users can pull the image using the complete URL
>
> ```bash
> $ podman pull docker.io/stig124/slidev:latest

> > ```

Reviewer: It looks like a more opinionated tools, which users could choose whatever they prefer, I don't think we need to document this here.

Author: podman is the drop-in replacement to Docker, deprecating Docker on some Enterprise Linux

Reviewer: like `yarn` to `npm`?

Author: I don't know the JS world, but pretty much yeah

Reviewer: Then I'd prefer just simply:

~~~md
```bash
$ docker pull stig124/slidev:latest
```

Or

```bash
$ podman pull docker.io/stig124/slidev:latest
```
~~~

Go into a directory with a `slides.md` file and run it :

```bash
$ docker run --rm -v .:/root/slides -p 3000:3030 stig124/slidev:latest
```

> You can change the port that will face outside by changing the number left of the port declaration => `3000:3030`

Open a browser on `localhost:3000` and you're ready to go, you can leave at any time using `Ctrl` + `C`

> > Please note that if you have a `node_modules` folder within the folder with the `slides.md` file, It will be **deleted** *(You can add `-e KEEP=1` next to `--rm` in the command to revert this behaviour)* as it may blocks the starting of the container

Reviewer: Why do we need to delete it? Is that possible to not mount it?

### Build the docker image

> If you want to build the image by yourself, you can :

Reviewer: ```suggestion
If you want to build the image by yourself from source, you can:
```

Author: This is not a build from source, just the Dockerfile is

Reviewer: Ah, then it's might not need to clone the main repo, otherwise this could introduce some confusing, as the npm installs might behavior differently than local source. Or we could move the script to another repo in the org that just contains those files.

Author: We can

Reviewer: Alright, what the common name of it? `slidevjs/dockfile`? I can create the repo and invite you

Author: I would prefer `container`, but `docker` or `image` would be a good name

Reviewer: https://github.com/antfu/container Here you go

Reviewer: Here you go https://github.com/slidevjs/container

Clone the repo

```bash
$ git clone https://github.com/slidevjs/slidev.git
```

> As the image downloads `slidev` from npm, you can also just download the `Dockerfile` and `entrypoint.sh` from the `Github` website

> Go to the folder where you have the `Dockerfile` and `entrypoint.sh` file

Reviewer: Feels a bit unnecessary as we just have them in the root. could just have this tree commands

```bash
$ git clone https://github.com/slidevjs/slidev.git
$ cd slidev
$ docker build -t <tag> .
```
